src/Electrical/Digital/logic_vectors.jl

```diff
@@ -0,0 +1,57 @@
+import Base: size, axes, getindex, setindex!
+
+const LogicOrNumber = Union{Logic, Number}
+
+struct StdULogicVector{N} <: AbstractArray{Logic, N}
+    logic::Array{Logic}
+    level::Array{Int}
+    function StdULogicVector(l::Array)
+        N = ndims(l)
+        l = Array{Logic}(convert.(Logic, l))
+        new{N}(l, get_logic_level.(l))
+    end
+end
+
+struct StdLogicVector{N} <: AbstractArray{Logic, N}
+    logic::Array{Logic}
+    level::Array{Int}
+    function StdLogicVector(l::Array)
+        N = ndims(l)
+        l = Array{Logic}(convert.(Logic, l))
+        new{N}(l, get_logic_level.(l))
+    end
+end
+
+const LogicVector = Union{StdULogicVector, StdLogicVector}
+
+Base.size(l::LogicVector) = size(l.logic)
+
+Base.axes(l::LogicVector) = axes(l.logic)
+
+function Base.getindex(s::LogicVector,
+                       i::Int)
+    getindex(s.logic, i)
+end
+function Base.getindex(s::LogicVector, i1::Int, i2::Int,
```

Comment:
Why do we call it a vector if it can be multidimensional?

Reply:
To remain consistent with `std_logic_vector` and `std_ulogic_vector` are the names used in IEEE Std_logic_1164.
Although its mostly used in vector form, it can be a matrix as well.